specification/communication/resource-manager/Microsoft.Communication/preview/2020-08-20-preview/CommunicationService.json

```diff
@@ -0,0 +1,1034 @@
+          },
+          "202": {
+            "description": "The delete operation is being performed in the background.",
+            "headers": {
+              "location": {
+                "description": "URL to query for status of the operation.",
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "The resource did not exist."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Delete resource": {
+            "$ref": "./examples/delete.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Communication/communicationServices/{communicationServiceName}/listKeys": {
+      "post": {
+        "description": "Get the access keys of the CommunicationService resource.",
+        "operationId": "CommunicationService_ListKeys",
+        "summary": "List Keys",
+        "tags": [ "CommunicationService" ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/CommunicationServiceName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The response describes CommunicationService access keys.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationServiceKeys"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "produces": [ "application/json" ],
+        "x-ms-examples": {
+          "List keys": {
+            "$ref": "./examples/listKeys.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Communication/communicationServices/{communicationServiceName}/regenerateKey": {
+      "post": {
+        "description": "Regenerate CommunicationService access key. PrimaryKey and SecondaryKey cannot be regenerated at the same time.",
+        "operationId": "CommunicationService_RegenerateKey",
+        "summary": "Regenerate Key",
+        "tags": [ "CommunicationService" ],
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "Parameter that describes the Regenerate Key Operation.",
+            "required": false,
```

Comment:
Do you intend this key type to be optional and take on default of 'primary'?

Reply:
didn't intend to, but will go with this for now to avoid another sdk generation